extmod/modplatform.c

```diff
@@ -0,0 +1,285 @@
+    return MP_OBJ_FROM_PTR(&info_platform_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(platform_platform_obj, 0, platform_platform);
+
+STATIC mp_obj_t platform_processor(void) {
+    return MP_OBJ_FROM_PTR(&info_processor_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_processor_obj, platform_processor);
+
+STATIC mp_obj_t platform_python_build(void) {
+    return MP_OBJ_FROM_PTR(&info_python_build_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_build_obj, platform_python_build);
+
+STATIC mp_obj_t platform_python_compiler(void) {
+    return MP_OBJ_FROM_PTR(&info_python_compiler_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_compiler_obj, platform_python_compiler);
+
+STATIC mp_obj_t platform_python_branch(void) {
+    return MP_OBJ_FROM_PTR(&info_python_branch_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_branch_obj, platform_python_branch);
+
+STATIC mp_obj_t platform_python_implementation(void) {
+    return MP_OBJ_FROM_PTR(&info_python_implementation_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_implementation_obj, platform_python_implementation);
+
+STATIC mp_obj_t platform_python_revision(void) {
+    return MP_OBJ_FROM_PTR(&info_python_revision_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_revision_obj, platform_python_revision);
+
+STATIC mp_obj_t platform_python_version(void) {
+    return MP_OBJ_FROM_PTR(&info_python_version_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_version_obj, platform_python_version);
+
+STATIC mp_obj_t platform_python_version_tuple(void) {
+    return MP_OBJ_FROM_PTR(&info_python_version_tuple_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_python_version_tuple_obj, platform_python_version_tuple);
+
+STATIC mp_obj_t platform_release(void) {
+    return MP_OBJ_FROM_PTR(&info_release_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_release_obj, platform_release);
+
+STATIC mp_obj_t platform_system(void) {
+    return MP_OBJ_FROM_PTR(&info_system_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_system_obj, platform_system);
+
+STATIC mp_obj_t platform_system_alias(void) {
+    return MP_OBJ_FROM_PTR(&info_system_alias_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_system_alias_obj, platform_system_alias);
+
+STATIC mp_obj_t platform_version(void) {
+    return MP_OBJ_FROM_PTR(&info_version_obj);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(platform_version_obj, platform_version);
+
+STATIC const qstr platform_uname_fields[] = {
+    MP_QSTR_system,
+    MP_QSTR_node,
+    MP_QSTR_release,
+    MP_QSTR_version,
+    MP_QSTR_machine,
+    MP_QSTR_processor
+};
+
+STATIC MP_DEFINE_ATTRTUPLE(
+    platform_uname_attr_obj,
+    platform_uname_fields,
+    6,
+    MP_ROM_PTR(&info_system_obj),
+    MP_ROM_PTR(&info_node_obj),
+    MP_ROM_PTR(&info_release_obj),
+    MP_ROM_PTR(&info_version_obj),
+    MP_ROM_PTR(&info_machine_obj),
+    MP_ROM_PTR(&info_processor_obj)
+    );
```

Comment:
This looks exactly the same as `os.uname()`, in which case the code can be reused... but actually it might make more sense for `os.uname()` to use this code here, so that it's centralised (ie doesn't need to be repeated per port).  In that case it can be done later, ie `os.uname()` points to here.

Reply:
Yes I agree.

Reply:
Actually not exactly the same, note this one adds an extra attribute (processor).